docs/source/examples/e2e_flow.rst

```diff
@@ -0,0 +1,438 @@
+.. _e2e_flow:
+
+==================================
+End-to-End Workflow with TorchTune
+==================================
+
+In this tutorial, we'll walk through an end-to-end example of how you can fine-tune,
+evaluate, optionally quantize and then run generation with your favorite LLM using
+TorchTune. We'll also go over how you can use some of your favorite tools and libraries
+from the community seemlessly with TorchTune.
+
+.. grid:: 2
+
+    .. grid-item-card:: :octicon:`mortar-board;1em;` What this tutorial will cover:
+
+      * Different type of recipes available in TorchTune beyond fine-tuning
+      * End-to-end example connecting all of these recipes
+      * Different tools and libraries you can use with TorchTune
+
+    .. grid-item-card:: :octicon:`list-unordered;1em;` Prerequisites
+
+      * Be familiar with the :ref:`overview of TorchTune<overview_label>`
+      * Make sure to :ref:`install TorchTune<install_label>`
+      * Concepts such as :ref:`configs <config_tutorial_label>` and checkpoints
+
+
+Overview
+--------
+
+Fine-tuning an LLM is almost never itself the end goal. Usually this is one step in a much
+larger worfklow. An example workflow might look something like this:
+
+- Download a popular model from HF Hub
+- Fine-tune the model using a relevant fine-tuning technique. The exact technique used
+  will depend on factors such as the model, amount and nature of training data, your hardware
+  setup and the end task for which the model will be used
+- Evaluate the model on some benchmarks to validate model quality
+- Run some generations to make sure the model output looks reasonable
+- Quantize the model for efficient inference followed by optionally exporting it for specific
+  environments such as inference on a mobile phone
+
+In this tutorial we'll cover each of these items and give examples of how you can do this using
+TorchTune, and how TorchTune makes it easy to use popular tools and libraries from the ecosystem.
+
+We'll use the Llama2 7B model for this tutorial. You can find a complete set of models supported
+by TorchTune `here <https://github.com/pytorch/torchtune/blob/main/README.md#introduction>`_.
```

Comment:
We should change this eventually. The official set of models should live in the docs.

Reply:
Also not seeing Gemma2b in that link.

Reply:
Yeh we need to update the README to include Gemma2 - thats next on deck!

Reply:
I can add it here - #668